spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClientQueryTagsTests.java

```diff
@@ -0,0 +1,215 @@
+ *
+ * @author Chris Bono
+ */
+class ConsulDiscoveryClientQueryTagsTests {
+
+	private static final String NAME = "query-tags-test-services";
+	static NewService QA_WEST_SERVICE = serviceForEnvironmentAndRegion("qa", "us-west", 9080);
+	static NewService QA_EAST_SERVICE = serviceForEnvironmentAndRegion("qa", "us-east", 9080);
+	static NewService PROD_WEST_SERVICE = serviceForEnvironmentAndRegion("prod", "us-west", 9082);
+	static NewService PROD_EAST_SERVICE = serviceForEnvironmentAndRegion("prod", "us-east", 9082);
+
+	private ApplicationContextRunner appContextRunner = new ApplicationContextRunner()
+			.withInitializer(new ConsulTestcontainers()).withConfiguration(AutoConfigurations.of(TestConfig.class))
+			.withPropertyValues("spring.application.name=consulServiceQueryTags",
+					"spring.cloud.consul.discovery.catalogServicesWatch.enabled=false");
+
+	private static NewService serviceForEnvironmentAndRegion(String env, String region, int port) {
+		NewService service = new NewService();
+		service.setAddress("localhost");
+		service.setId(String.format("%s-%s-%s", NAME, env, region));
+		service.setName(NAME);
+		service.setPort(port);
+		service.setTags(Arrays.asList(env, region));
+		return service;
+	}
+
+	@Test
+	void singleMatchingTagSpecifiedOnDefaultQueryTagProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=qa").run(
+				context -> assertThatGetInstancesReturnsExpectedServices(context, QA_WEST_SERVICE, QA_EAST_SERVICE));
+	}
+
+	@Test
+	void singleNonMatchingTagSpecifiedOnDefaultQueryTagProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=foo")
+				.run(context -> assertThatGetInstancesReturnsExpectedServices(context, new NewService[0]));
+	}
+
+	@Test
+	void multipleMatchingTagsSpecifiedOnDefaultQueryTagProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=prod,us-west")
+				.run(context -> assertThatGetInstancesReturnsExpectedServices(context, PROD_WEST_SERVICE));
+	}
+
+	@Test
+	void multipleNonMatchingTagsSpecifiedOnDefaultQueryTagProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=prod,foo")
+				.run(context -> assertThatGetInstancesReturnsExpectedServices(context, new NewService[0]));
+	}
+
+	@Test
+	void multipleConflictingMatchingTagsSpecifiedOnDefaultQueryTagsProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=prod,qa")
+				.run(context -> assertThatGetInstancesReturnsExpectedServices(context));
+	}
+
+	@Test
+	void emptyTagSpecifiedOnDefaultQueryTagProperty() {
+		appContextRunner.withPropertyValues("spring.cloud.consul.discovery.default-query-tag=")
+				.run(context -> assertThatGetInstancesReturnsExpectedServices(context, QA_WEST_SERVICE, QA_EAST_SERVICE,
+						PROD_WEST_SERVICE, PROD_EAST_SERVICE));
+	}
+
+	@Test
+	void singleTagSpecifiedOnServerListQueryTagsProperty() {
```

Comment:
I attempted to reduce overlap w/ the `ConsulDiscoverPropertiesTest` as much as possible. However there is still overlap. 

One approach to reduce the overlap would be to mock out the `CDP.getQueryTagsForService` to return  whatever is needed rather than have the same group of tests for the default query tag and the server list query tag values. Ideally, this test should **not** concern itself w/ the mechanics of `CDP#getQueryTagsForService`. That being said, there is still value in having these here as they do integration test the `@ConfigurationProperties` aspect of CDP. 

I realized I was bikeshedding and when I do that while writing tests, I typically err on the side of extra coverage rather than thinning it out.